jolpica_api/ergastapi/tests/test_views.py

```diff
@@ -33,7 +33,22 @@ def test_viewsets(client: APIClient, endpoint: str, path: Path, django_assert_ma
         response = client.get(f"/ergast/f1/{endpoint}")
     assert response.status_code == 200
 
+    # fetch the same data without the .json suffix and ensure that it matches exactly,
+    # except for the difference in the URL itself
+    # this is necessessary because the router uses complicated regexes to match the endpoint
+    # which may cause differences between the two responses if not handled correctly
+    html_response = client.get(f"/ergast/f1/{endpoint}".replace(".json", ""), follow=True)
+    # we need follow=True because the non-json endpoint will redirect from no trailing slash to trailing slash
```

Comment:
```suggestion
    html_response = client.get(f"/ergast/f1/{endpoint}".replace(".json", "/"), follow=True)
    # we need follow=True because the non-json endpoint will redirect from no trailing slash to trailing slash
```
Would this change remove the need for a trailing slash?

Reply:
Yes, but technically it's a different test case then. And for parity, we'd want the endpoint without trailing slash to work. I can see it being possible to break one but not the other with the regex stuff going on.